Resources/Textures/Shaders/radial-menu.swsl

```diff
@@ -0,0 +1,162 @@
+
+    //angle of the line
+    highp vec2 d = FRAGCOORD.xy - centerPos;
+    highp float angle = getAngle(d);
+
+    highp float r = length(FRAGCOORD.xy - centerPos);
+    // fill sectors
+    if(r > innerRadius && r < outerRadius)
+    {
+        highp float g = fillGradient(r, innerRadius, outerRadius);
+
+        // trying to mix in color per button
+
+        highp float from = 0;
+        for (int i = 0; i < childCount; i++)
+        {
+            highp float to = separatorAngles[i];
+            if(angle > from && angle < to)
+            {
+                col +=  vec4(sectorColors[i].xyz, g);;
+            }
+
+            from = to;
+        }
+    }
+
+    // get step of radial menu buttons in radian
+    highp float halfSectorAngleSize = (separatorAngles[1] - separatorAngles[0]) * 0.5;
+
+    for (int i = 0; i < childCount; i++)
+    {
+        highp float sectorMedian = sectorMedianAngles[i];
+        highp float sectorMedianToAngleDiff = abs(sectorMedian - angle);
+        highp vec4 borderColor = borderColors[i];
+        highp vec4 borderColorLight = borderColor * 0.6;
+        if(r > innerRadius)
+        {
+            highp float iAngle = twopi - separatorAngles[i];
+            // button separators
+            highp float sectorFromAngle;
+            if(i > 0)
+            {
+                sectorFromAngle = separatorAngles[i - 1];
+            }
+            else
+            {
+                sectorFromAngle = 0;
+            }
+
+            col += separator(d, r, outerRadius, sectorFromAngle, 0.5) * borderColor;
+            col += separator(d, r, outerRadius, iAngle, 0.5) * borderColor;
+
+            // set up decorations
+            // inner button 'square' decoration
+            if(r > innerRadius + 15)
+            {
+                col += separator(d, r, outerRadius - 15, twopi - sectorMedian - halfSectorAngleSize * 0.8, 1.0) * 0.2 * borderColorLight;
+                col += separator(d, r, outerRadius - 15, twopi - sectorMedian + halfSectorAngleSize * 0.8, 1.0) * 0.2 * borderColorLight;
+
+                if(sectorMedianToAngleDiff < halfSectorAngleSize * 0.8)
+                {
+                    col += circle(r, innerRadius + 15, 2.0) * 0.2 * borderColorLight;
+                    col += circle(r, outerRadius - 15, 2.0) * 0.2 * borderColorLight;
+                }
+            }
+        }
+
+        // outer button decorative elements
+        if(sectorMedianToAngleDiff < halfSectorAngleSize * 0.2)
+        {
+            col += circle(r, innerRadius - 5, 4.0) * 0.4 * borderColor;
+        }
+
+        if(sectorMedianToAngleDiff < halfSectorAngleSize * 0.6)
+        {
+            col += circle(r, outerRadius + 10, 4.0) * 0.4 * borderColor;
+        }
+
+        // outer and inner circle of sectors
+        if(sectorMedianToAngleDiff < halfSectorAngleSize)
+        {
+            col += circle(r, innerRadius, 2.0) * borderColor;
+            col += circle(r, outerRadius, 2.0) * borderColor;
+        }
```

Comment:
Are you sure you even need these if statements and can't just inline the variables?